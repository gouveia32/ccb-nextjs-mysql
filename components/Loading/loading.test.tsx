import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { Loading, LoadingProps } from "./loading.component";
import { CircularProgress } from "@material-ui/core";

describe("Loading component", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    const mockProps: LoadingProps = {
      size: 10,
      classname: "",
    };

    wrapper = shallow(<Loading {...mockProps} />);
  });

  it("should render Loading component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render CircularProgress component with correct size prop", () => {
    const circularProgress = wrapper.find(CircularProgress);
    expect(circularProgress).toHaveLength(1);
    expect(circularProgress.prop("size")).toBe(10);
  });
});
