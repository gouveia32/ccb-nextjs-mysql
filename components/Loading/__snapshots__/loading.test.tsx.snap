// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render Loading component 1`] = `ShallowWrapper {}`;
